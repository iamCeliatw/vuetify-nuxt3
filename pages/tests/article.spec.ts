import { shallowMount } from "@vue/test-utils";
import Article from "../article/[key].vue";
vi.stubGlobal(
  "useHead",
  vi.fn(() => ({ title: "Article" }))
);
vi.stubGlobal("useRouter", () => ({}));

const mockUseRoute = () => {
  //params
  vi.stubGlobal(
    "useRoute",
    vi.fn(() => ({
      params: {
        key: ["test"],
      },
    }))
  );
};
const mockUseAsyncData = (data: any) => {
  vi.stubGlobal(
    "useAsyncData",
    vi.fn(() => ({
      data: ref([{ title: "Test Title", description: "Test Description" }]),
      pending: ref(false),
      error: ref(null),
    }))
  );
};

const mockUseSupabaseClient = (data: any) => {
  vi.stubGlobal(
    "useSupabaseClient",
    vi.fn(() => ({
      data: ref(data),
      pending: ref(false),
      error: ref(null),
      from: vi.fn(() => ({
        select: vi.fn(() => ({
          equal: vi.fn(() =>
            Promise.resolve({
              equal: vi.fn(() =>
                Promise.resolve({
                  data: [
                    {
                      id: 1,
                      title: "Test Article",
                      description: "This is a test article",
                    },
                  ],
                  error: null,
                })
              ),
            })
          ),
        })),
      })),
    }))
  );
};

describe("Article", () => {
  beforeEach(() => {
    vi.resetAllMocks();
    mockUseSupabaseClient([]);
    mockUseRoute();
    mockUseAsyncData([]);
  });
  it("should render correctly", () => {
    const wrapper = shallowMount(Article);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("useHead should be called with specific object", () => {
    const wrapper = shallowMount(Article);
    expect(useHead).toHaveBeenCalledWith(
      expect.objectContaining({
        title: "Test Title",
        meta: [
          {
            property: "og:title",
            content: "Test Title",
          },
          {
            property: "og:description",
            content: "Test Description",
          },
        ],
      })
    );
  });
});
