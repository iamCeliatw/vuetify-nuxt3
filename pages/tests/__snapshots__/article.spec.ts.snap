// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Article > should render correctly 1`] = `
"<navbar-stub data-v-a3507ac1=""></navbar-stub>
<section data-v-a3507ac1="" class="article__wrapper">
  <div data-v-a3507ac1="" class="article__container">
    <content-stub data-v-a3507ac1="" articledata="[object Object]"></content-stub>
  </div>
</section>"
`;
